src/deepsparse/server/build_logger.py

```diff
@@ -111,42 +111,93 @@ def build_leaf_loggers(
+    Build a set of data loggers (FunctionLogger instances)
+    according to the configuration.
 
     :param endpoints: A list of server's endpoint configurations;
         the configurations contain the information about the metric
         functions (MetricFunctionConfig objects)
         and the targets that the functions are to be applied to
     :param loggers: The created "leaf" loggers
-    :return: A list of FunctionLogger instances
+    :return: A list of FunctionLogger instances responsible
+        for logging data information
     """
-    function_loggers = []
+    data_loggers = []
     for endpoint in endpoints:
         if endpoint.data_logging is None:
             continue
         for target, metric_functions in endpoint.data_logging.items():
-            target_identifier = _get_target_identifier(endpoint.name, target)
+            target_identifier = _get_target_identifier(target, endpoint.name)
             for metric_function in metric_functions:
-                function_loggers.append(
+                data_loggers.append(
                     _build_function_logger(metric_function, target_identifier, loggers)
                 )
-    return function_loggers
-
-
-def _create_system_logger(loggers: Dict[str, BaseLogger]) -> FunctionLogger:
-    # returns a function logger that matches to all system logs, logging
-    # every system call to each leaf logger
-    return _build_function_logger(
-        metric_function_cfg=MetricFunctionConfig(
-            func="identity", frequency=1, target_loggers=None
-        ),
-        target_identifier=f"category:{MetricCategories.SYSTEM.value}",
-        loggers=loggers,
-    )
+    return data_loggers
+
+
+def build_system_loggers(
+    loggers: Dict[str, BaseLogger], system_logging_config: SystemLoggingConfig
+) -> List[FunctionLogger]:
+    """
+    Create a system loggers according to the configuration specified
+    in `system_logging_config`. System loggers are FunctionLogger instances
+    responsible for logging system groups metrics.
+
+    :param loggers: The created "leaf" loggers
+    :param system_logging_config: The system logging configuration
+    :return: A list of FunctionLogger instances responsible for logging system data
+    """
+    system_loggers = []
+    if not system_logging_config.enable:
+        return system_loggers
+
+    for config_group_name, config_group_args in system_logging_config:
+        if not isinstance(config_group_args, SystemLoggingGroup):
+            continue
+        if not config_group_args.enable:
+            continue
+
+        system_loggers.append(
+            _build_function_logger(
+                metric_function_cfg=MetricFunctionConfig(
+                    func="identity",
+                    frequency=1,
+                    target_loggers=config_group_args.target_loggers,
+                ),
+                target_identifier=_get_target_identifier(config_group_name),
```

Comment:
what does `_get_target_identifier` do here?
and don't we need to update the identifier to match to all logged values containing the config group name?

Reply:
In this case, it acts as an "identity" function. But I wanted to keep it here in case we were to add functionalities in the future (e.g. this function handles making sure that regex identifiers will not be modified).
There was no need to update the identifier while the `identifier_pre_fix` was present in the `pipeline`. 

Let's sync offline and discuss, I think this is an important point we need to align on.